http://sourceware.org/ml/gdb-patches/2016-01/msg00137.html
Subject: [testsuite patch] Fix gdb.multi/base.exp testsuite regression  [Re: [ob/pushed] Stop using nowarnings in gdb/testsuite/gdb.multi/]


--fdj2RfSjLxBAspz7
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

On Thu, 10 Dec 2015 17:31:11 +0100, Pedro Alves wrote:
> Several of the gdb.multi tests use the "nowarnings" option to suppress
> warnings.

commit 762f774785f4ef878ac4c831e1f4733dc957234d
Author: Pedro Alves <palves@redhat.com>
Date:   Thu Dec 10 16:21:06 2015 +0000
    Stop using nowarnings in gdb/testsuite/gdb.multi/

+gdb compile failed, gdb/testsuite/gdb.multi/hello.c: In function 'commonfun':
+gdb/testsuite/gdb.multi/hello.c:24:19: warning: implicit declaration of function 'bar' [-Wimplicit-function-declaration]
+ int commonfun() { bar(); } /* from hello */
+                   ^
+gdb/testsuite/gdb.multi/hello.c: At top level:
+gdb/testsuite/gdb.multi/hello.c:26:1: warning: return type defaults to 'int' [-Wimplicit-int]
+ bar()
+ ^
+gdb/testsuite/gdb.multi/hello.c:32:1: warning: return type defaults to 'int' [-Wimplicit-int]
+ hello(int x)
+ ^
+gdb/testsuite/gdb.multi/hello.c:38:1: warning: return type defaults to 'int' [-Wimplicit-int]
+ main()
+ ^
+UNTESTED: gdb.multi/base.exp: base.exp


OK for check-in?


Jan

--fdj2RfSjLxBAspz7
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline; filename=1

gdb/testsuite/ChangeLog
2016-01-08  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* gdb.multi/goodbye.c: Fix compilation warnings by adding return types
	and reordering the functions.
	* gdb.multi/hangout.c: Likewise.
	* gdb.multi/hello.c: Likewise.

diff --git a/gdb/testsuite/gdb.multi/goodbye.c b/gdb/testsuite/gdb.multi/goodbye.c
index 701e504..bf7759a 100644
--- a/gdb/testsuite/gdb.multi/goodbye.c
+++ b/gdb/testsuite/gdb.multi/goodbye.c
@@ -39,24 +39,28 @@ int verylongfun()
   glob *= 9;
 }
 
-main() {
-  mailand();
-  foo(glob);
-  verylongfun();
-  goodbye();
-}
-
-foo(int x) {
-  return x + 92;
-}
-
+void
 mailand()
 {
   glob = 46;
 }
 
-void commonfun() { mailand(); } /* from goodbye */
+int
+foo(int x) {
+  return x + 92;
+}
 
+void
 goodbye() {
   ++glob;
 }
+
+int
+main() {
+  mailand();
+  foo(glob);
+  verylongfun();
+  goodbye();
+}
+
+void commonfun() { mailand(); } /* from goodbye */
diff --git a/gdb/testsuite/gdb.multi/hangout.c b/gdb/testsuite/gdb.multi/hangout.c
index e2c41b8..3701512 100644
--- a/gdb/testsuite/gdb.multi/hangout.c
+++ b/gdb/testsuite/gdb.multi/hangout.c
@@ -17,6 +17,7 @@
 
 #include <stdio.h>
 
+int
 main(int argc, char *argv[])
 {
   int i;
diff --git a/gdb/testsuite/gdb.multi/hello.c b/gdb/testsuite/gdb.multi/hello.c
index efff59d..93d921f 100644
--- a/gdb/testsuite/gdb.multi/hello.c
+++ b/gdb/testsuite/gdb.multi/hello.c
@@ -21,20 +21,23 @@ short hglob = 1;
 
 short glob = 92;
 
-int commonfun() { bar(); } /* from hello */
-
+void
 bar()
 {
   if (glob == 0)
     exit(1);
 }
 
+int commonfun() { bar(); } /* from hello */
+
+int
 hello(int x)
 {
   x *= 2;
   return x + 45;
 }
 
+int
 main()
 {
   int tmpx;

--fdj2RfSjLxBAspz7--

